ts4.1/index.d.ts

```diff
@@ -0,0 +1,294 @@
+	 *
+	 * type T3 = Camelize<string>;
+	 * // => string
+	 *
+	 * type T4 = Camelize<"foo_bar", true>;
+	 * // => "FooBar"
+	 *
+	 * type T5 = Camelize<"foo_bar", boolean>;
+	 * // => "fooBar" | "FooBar"
+	 * ```
+	 */
+	type Camelize<S extends string, PascalCase extends boolean = false> =
+		S extends `_${infer S2}` ? Camelize<S2, PascalCase> :
+		S extends `-${infer S2}` ? Camelize<S2, PascalCase> :
+		S extends `.${infer S2}` ? Camelize<S2, PascalCase> :
+		S extends ` ${infer S2}` ? Camelize<S2, PascalCase> :
+		S extends `${infer S1}_${infer S2}` ? `${Camelize<S1, PascalCase>}${Camelize<S2, true>}` :
+		S extends `${infer S1}-${infer S2}` ? `${Camelize<S1, PascalCase>}${Camelize<S2, true>}` :
+		S extends `${infer S1}.${infer S2}` ? `${Camelize<S1, PascalCase>}${Camelize<S2, true>}` :
+		S extends `${infer S1} ${infer S2}` ? `${Camelize<S1, PascalCase>}${Camelize<S2, true>}` :
+		PascalCase extends true ? Capitalize<S> : S;
+
+	/**
+	 * @internal
+	 * Digs a set of paths by one depth. Used for processing `stopPaths`.
+	 *
+	 * @param Path a union of string literal types, each of which is a "path" from the root.
+	 * @param S a prefix word. A string literal type or `string`.
+	 *
+	 * @example
+	 * ```
+	 * type T1 = FilterPaths<"foo.bar" | "foo.baz" | "foo.bar.baz" | "other.path", "foo">;
+	 * // => "bar" | "baz" | "bar.baz"
+	 *
+	 * type T2 = FilterPaths<string, "foo">;
+	 * // => string
+	 *
+	 * type T3 = FilterPaths<"foo.bar" | "foo.baz" | "foo.bar.baz" | "other.path", string>;
+	 * // => string
+	 *
+	 * type T3 = FilterPaths<"foo" | "bar", string>;
+	 * // => never
+	 * ```
+	 */
+	type FilterPaths<Path extends string, S extends string> =
+		// We treat `string` as "indeterminate set of strings", so return `string` in this case.
+		string extends Path ? string :
+		// If `S` is indeterminate, we are not sure about the resulting set,
+		// except when there is no path containing `.`.
+		string extends S ?
+		Path extends `${string}.${string}` ? string : never :
+		// Otherwise, do a normal filtering.
+		// If a path starts with `${S}.`, then return the suffix.
+		// Otherwise return never (which means the candidate is rejected)
+		Path extends `${S}.${infer Suffix}` ? Suffix : never;
+
+	/**
+	 * @internal
+	 * Helper type to conditionally camelize a string.
+	 *
+	 * @param S a union of string literal types to camelcase.
+	 * @param PascalCase if true, the first word is also capitalized.
+	 * @param Exclude a union of string literal to exclude from camelization.
+	 *
+	 * @example
+	 * ```
+	 * type T1 = CamelcaseUnless<"foo_bar" | "bar_baz", false, never>;
+	 * // => "fooBar" | "barBaz"
+	 *
+	 * type T2 = CamelcaseUnless<"foo_bar" | "bar_baz", false, "bar_baz">;
+	 * // => "fooBar" | "bar_baz"
+	 *
+	 * type T1 = CamelcaseUnless<"foo_bar" | "bar_baz", false, string>;
+	 * // => "foo_bar" | "fooBar" | "bar_baz" | "barBaz"
+	 * ```
+	 */
+	type CamelcaseUnless<S extends string, PascalCase extends boolean, Exclude extends string> =
```

Comment:
```suggestion
	type CamelCaseUnless<S extends string, PascalCase extends boolean, Exclude extends string> =
```

Reply:
I just followed how `camelcaseKeys` is capitalized, but OK :+1:

Reply:
Yeah, I didn't realize "camel case" was not one word when I initially made this package.